### DISCUSSION TITLE
Make Floorp a private alternative to Vivaldi

### DISCUSSION THREAD

**Discussion by stilgarwolf** [Floorp-Projects/Floorp#1000] [Feature Request]
As Vivaldi cannot escape Chromium lack of privacy trap, I think some enhancements would make Floorp very attractive for old Opera and Vivaldi users (Floorp is also discussed in Vivaldi forums!):

1. Enhanced tab handling: allow user to read now or later pages without losing control of the page of origin. Extreme useful when reading documentation or news with many links.
 - In link menu: new "Open link in foreground tab" and "Open link in background tab" entries (as first entries?)
 - Close tab activation to allow activate tabs in recently used order.

 2. Speed dial. (I have more than 100 folders, subfolders and links in my current speed dial, apart of traditional bookmarks.)

 3. Tab stacking: selecting tabs to stack may enhance current split view. And, maybe, add synchronized scroll option.

 4. Add a "Clear data" toolbar button and/or shortcut to "Clear data..." button in Settings.

Others:
 - search selected text with selectable search engine
 - allow users to edit menus.
 - enhance "Reopen closed tab" button to let user choose which tab to reopen.

**Comment by CutterKnife** [comment]
1. Enhanced tab handling
	1. Holding down Ctrl (Cmd) and clicking on the link should open the tab in the background.
	2. Sorry, I did not understand what you meant. There is a setting in the tab group settings that allows Ctrl (Cmd) + Tab to switch tabs in the order most recently used.
3. I think there was an extensions that did that.
4. It has been considered, but was abandoned due to problems with the tab management APIs.
5. Keyboard Shortcuts are gradually increasing with updates. If it is technically feasible, they may become available in the future.

Others:
1. The menu in the address bar allows you to select a search engine to use only this time, but do you change engines frequently?
2. Interesting idea. I'm not a developer so IDK, but I remember that context menus are prone to OS compatibility issues.
3. It can be selected in the "Recently Closed Tabs" menu or in Floorp View.

In any case, they are very under-resourced. Even if it were reasonable and implementable, being able to get to work is another matter. If they implement it, they'll have to maintain it as well.
But I think it is an interesting ideas :)

**Reply by stilgarwolf** [reply]
1. Holding down Ctrl opens link in foreground, not in background, at least on my computer.
Tab handling is not about switching tabs but about (much more used) closing tabs: close current tab and go to previous **_used_** tab, not to the left tab. (see https://github.com/Floorp-Projects/Floorp/discussions/783)
2. There are many Speed Dial extensions, no one as capable as Vivaldi.
3. I don't think tab stacking is more complicated than tab splitting, but I think is more useful.

The others are not so important:
Vivaldi has "Search with XX engine" entries in select popup, with XXx taken from settings. For Firefox, a "Search with Wikipedia" extension exists. I almost never use Google, Yahoo or Bing, so my search results may vary by search engine.

Some time ago Firefox permitted extensions to edit context menu, but following Google example ("the user is dumb, we know better what he needs"), removed the permission due to "security considerations".

I know the team is under resourced, but I think some (as tab handling) are easy to implement.

Unfortunately, no customizable, secure and private browser exists for power users, (or I don't know one).
But power users brought users to the old Firefox and those same power users were pushed by the new Firefox to other browsers.
